classes/chat/ChatManager.php

```diff
@@ -142,6 +150,45 @@ private function fetchPosts(?int $starting_post_id = null, int $max_posts = self
             }
             $post->message = nl2br($this->system->html_parse($post->message, false, true));
 
+            // Handle Quotes
+            $pattern = "/\[quote:\d+\]/";
+            $has_quote = preg_match_all($pattern, $post->message, $matches);
+            if ($has_quote) {
+                if ($allow_quote) {
+                    foreach ($matches[0] as $match) {
+                        // get chat post contents
+                        $pattern = "/\[quote:(\d+)\]/";
+                        preg_match($pattern, $match, $id);
+                        $quote = $this->fetchPosts($id[1], 1, false);
```

Comment:
```suggestion
                        $quote = $this->fetchPost($id[1]);
```
It gets a little contrived using the batch method to fetch one post, we should add a helper method like this or just inline the query here.

Reply:
Marking as to-do.